source/module_hsolver/diago_pexsi.cpp

```diff
@@ -0,0 +1,22 @@
+#include "c_pexsi_interface.h"
+#include "diago_pexsi.h"
+#include "module_base/global_variable.h"
+#include "module_base/lapack_connector.h"
+#include "module_base/timer.h"
+#include "module_base/tool_quit.h"
+
+namespace hsolver
+{
+
+void DiagoPexsi::diag(hamilt::Hamilt<double>* phm_in, psi::Psi<double>& psi, double* eigenvalue_in)
```

Comment:
This class is only a starting point for PEXSI interface, you should add UnitTest for this class in this PR. In the future, you should update code and UT at next PRs at the same time.